### PR TITLE
Cleanup/java object factory

### PR DESCRIPTION
**Pull request by owen-jones-diffblue**
I've been reading this code as we're making an equivalent for C. I noticed a few things that could be tidied up. There are no functional changes.

### CODE REVIEW

**Comment by owen-jones-diffblue** [comment]
@mgudemann You seem to have written quite a lot of the code in this file, so you might be a good person to review. On the other hand you seem to be on holiday next week.

File: src/java_bytecode/java_object_factory.cpp [hunk 1]
```diff
@@ -582,15 +582,18 @@ exprt object_factory(
...
-    gen_nondet_init(
-      object,
+    java_object_factoryt state(
       init_code,
+      !allow_null,
+      max_nondet_array_length,
       symbol_table,
-      loc,
+      message_handler);
+    state.gen_nondet_init_rec(
```
Comment (peterschrammel) [inline src/java_bytecode/java_object_factory.cpp]: Was the renaming necessary then?

Comment (owen-jones-diffblue) [inline src/java_bytecode/java_object_factory.cpp]: No, I guess not. It was useful to be certain that the function was okay to remove. I'll add a commit to rename the remaining function back.

Comment (tautschnig) [inline src/java_bytecode/java_object_factory.cpp]: You might just drop that earlier commit and rebase?

**Comment by owen-jones-diffblue** [comment]
I've dropped the renaming commit.

**Pull request merged**